packages/datadog-instrumentations/src/helpers/hook.js

```diff
@@ -0,0 +1,41 @@
+'use strict'
+
+const path = require('path')
+const iitm = require('../../../dd-trace/src/iitm')
+const ritm = require('../../../dd-trace/src/ritm')
+
+function Hook (modules, onrequire) {
+  if (!(this instanceof Hook)) return new Hook(modules, onrequire)
+
+  this._patched = {}
+
+  const safeHook = (moduleExports, moduleName, moduleBaseDir) => {
+    const parts = [moduleBaseDir, moduleName].filter(v => v)
+    const filename = path.join(...parts)
+
+    if (this._patched[filename]) return moduleExports
+
+    this._patched[filename] = true
+
+    return onrequire(moduleExports, moduleName, moduleBaseDir)
+  }
+
+  this._ritmHook = ritm(modules, {}, safeHook)
+  this._iitmHook = iitm(modules, {}, (moduleExports, moduleName, moduleBaseDir) => {
+    // Default export is always moved to `default` for CommonJS modules.
```

Comment:
Let's add a TODO comment to implement this in IITM

Reply:
This rule only applies to us specifically, not sure it should go in iitm unless the goal of iitm changes to only support CommonJS modules loaded through ESM loader hooks.

Reply:
[It can, yeah.](https://github.com/DataDog/dd-trace-js/pull/1934#discussion_r836702221) Right now we're not dependent on that, but we do something there if needed.